### ISSUE TITLE
Syntactically invalid try … catch block in eslint.js

### ISSUE THREAD

**Issue by bakert** [eslint/eslint#13352]
There's a bare `try { … } catch { … }` in 7.1.0's eslint.js. It's not valid js to have a try/catch block without an error variable after the catch as in `try { … } catch (error_var) { … }`. This causes eslint to break in some places (but not all).

**Tell us about your environment**

Node 11 on heroku via Travis. 
Also found invalid by https://esprima.org/demo/validate.html
Also breaks the spec found at http://es5.github.io/#x12.14

* **ESLint Version:** 7.1.0
* **Node Version:** 11.0.0
* **npm Version:** 

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    "parser": "babel-eslint",
    "parserOptions": {
        "sourceType": "module",
        "allowImportExportEverywhere": true
    },
    "env": {
        "browser": true,
        "es6": true
    },
    "extends": "eslint:recommended",
    "rules": {
        "accessor-pairs": "error",
        "array-bracket-newline": "off",
        "array-bracket-spacing": [
            "error",
            "never"
        ],
        "array-callback-return": "error",
        "array-element-newline": "off",
        "arrow-body-style": "error",
        "arrow-parens": "error",
        "arrow-spacing": "error",
        "block-scoped-var": "error",
        "block-spacing": [
            "error",
            "always"
        ],
        "brace-style": "off",
        "callback-return": "error",
        "camelcase": "error",
        "capitalized-comments": "off",
        "class-methods-use-this": "error",
        "comma-dangle": "error",
        "comma-spacing": "off",
        "comma-style": [
            "error",
            "last"
        ],
        "complexity": "error",
        "computed-property-spacing": [
            "error",
            "never"
        ],
        "consistent-return": "off",
        "consistent-this": "error",
        "curly": "off",
        "default-case": "error",
        "dot-location": ["error", "property"],
        "dot-notation": "off",
        "eol-last": "error",
        "eqeqeq": "error",
        "for-direction": "error",
        "func-call-spacing": "error",
        "func-name-matching": "error",
        "func-names": [
            "error",
            "never"
        ],
        "func-style": "error",
        "function-paren-newline": "error",
        "generator-star-spacing": "error",
        "getter-return": "error",
        "global-require": "error",
        "guard-for-in": "error",
        "handle-callback-err": "error",
        "id-blacklist": "error",
        "id-length": "off",
        "id-match": "error",
        "implicit-arrow-linebreak": "error",
        "indent": "error",
        "indent-legacy": "error",
        "init-declarations": "off",
        "jsx-quotes": "error",
        "key-spacing": "off",
        "keyword-spacing": "off",
        "line-comment-position": "off",
        "linebreak-style": [
            "error",
            "unix"
        ],
        "lines-around-comment": "off",
        "lines-around-directive": "error",
        "lines-between-class-members": "error",
        "max-depth": "error",
        "max-len": "off",
        "max-lines": [
            "error",
            650
        ],
        "max-nested-callbacks": "error",
        "max-params": "off",
        "max-statements": "off",
        "max-statements-per-line": "off",
        "multiline-comment-style": "off",
        "new-cap": "error",
        "new-parens": "error",
        "newline-after-var": "off",
        "newline-before-return": "off",
        "newline-per-chained-call": "off",
        "no-alert": "error",
        "no-array-constructor": "error",
        "no-await-in-loop": "error",
        "no-bitwise": "error",
        "no-buffer-constructor": "error",
        "no-caller": "error",
        "no-catch-shadow": "error",
        "no-confusing-arrow": "error",
        "no-console": "error",
        "no-continue": "error",
        "no-div-regex": "error",
        "no-duplicate-imports": "error",
        "no-else-return": "off",
        "no-empty-function": "error",
        "no-eq-null": "off",
        "no-eval": "error",
        "no-extend-native": "error",
        "no-extra-bind": "error",
        "no-extra-label": "error",
        "no-extra-parens": "off",
        "no-floating-decimal": "error",
        "no-global-assign": ["error", { "exceptions": ["Deckbox"] }],
        "no-implicit-globals": "off",
        "no-implied-eval": "error",
        "no-inline-comments": "off",
        "no-inner-declarations": [
            "error",
            "functions"
        ],
        "no-invalid-this": "off",
        "no-iterator": "error",
        "no-label-var": "error",
        "no-labels": "error",
        "no-lone-blocks": "error",
        "no-lonely-if": "error",
        "no-loop-func": "error",
        "no-magic-numbers": "off",
        "no-mixed-operators": "off",
        "no-mixed-requires": "error",
        "no-multi-assign": "error",
        "no-multi-spaces": "off",
        "no-multi-str": "error",
        "no-multiple-empty-lines": "error",
        "no-native-reassign": "off", // Deprecated. See: https://eslint.org/docs/rules/no-native-reassign
        "no-negated-condition": "off",
        "no-negated-in-lhs": "error",
        "no-nested-ternary": "error",
        "no-new": "error",
        "no-new-func": "error",
        "no-new-object": "error",
        "no-new-require": "error",
        "no-new-wrappers": "error",
        "no-octal-escape": "error",
        "no-param-reassign": "off",
        "no-path-concat": "error",
        "no-plusplus": [
            "error", {
                "allowForLoopAfterthoughts": true
            }
        ],
        "no-process-env": "error",
        "no-process-exit": "error",
        "no-proto": "error",
        "no-prototype-builtins": "error",
        "no-restricted-globals": "error",
        "no-restricted-imports": "error",
        "no-restricted-modules": "error",
        "no-restricted-properties": "error",
        "no-restricted-syntax": "error",
        "no-return-assign": "off",
        "no-return-await": "error",
        "no-script-url": "error",
        "no-self-compare": "error",
        "no-sequences": "error",
        "no-shadow": "error",
        "no-shadow-restricted-names": "error",
        "no-spaced-func": "error",
        "no-sync": "error",
        "no-tabs": "error",
        "no-template-curly-in-string": "error",
        "no-ternary": "off",
        "no-throw-literal": "error",
        "no-trailing-spaces": "error",
        "no-undef-init": "error",
        "no-undefined": "error",
        "no-underscore-dangle": "off",
        "no-unmodified-loop-condition": "error",
        "no-unneeded-ternary": "error",
        "no-use-before-define": "error",
        "no-useless-call": "error",
        "no-useless-computed-key": "error",
        "no-useless-concat": "off",
        "no-useless-constructor": "error",
        "no-useless-rename": "error",
        "no-useless-return": "error",
        "no-var": "off",
        "no-void": "error",
        "no-warning-comments": "error",
        "no-whitespace-before-property": "error",
        "no-with": "error",
        "nonblock-statement-body-position": "error",
        "object-curly-newline": "error",
        "object-curly-spacing": "off",
        "object-shorthand": ["error", "properties"],
        "one-var": "off",
        "one-var-declaration-per-line": "off",
        "operator-assignment": [
            "error",
            "always"
        ],
        "operator-linebreak": "error",
        "padded-blocks": "off",
        "padding-line-between-statements": "error",
        "prefer-arrow-callback": "off",
        "prefer-const": "error",
        "prefer-destructuring": "off",
        "prefer-numeric-literals": "error",
        "prefer-promise-reject-errors": "error",
        "prefer-reflect": "off",
        "prefer-rest-params": "error",
        "prefer-spread": "error",
        "prefer-template": "off",
        "quote-props": "off",
        "quotes": [
            "error",
            "double",
            {
                "avoidEscape": true
            }
        ],
        "radix": [
            "error",
            "always"
        ],
        "require-await": "error",
        "require-jsdoc": "off",
        "rest-spread-spacing": "error",
        "semi": "error",
        "semi-spacing": [
            "error", {
                "after": true,
                "before": false
            }
        ],
        "semi-style": [
            "error",
            "last"
        ],
        "sort-imports": "error",
        "sort-keys": "off",
        "sort-vars": "off",
        "space-before-blocks": "error",
        "space-before-function-paren": "off",
        "space-in-parens": "off",
        "space-infix-ops": "off",
        "space-unary-ops": "off",
        "spaced-comment": "off",
        "strict": [
            "error",
            "never"
        ],
        "switch-colon-spacing": "error",
        "symbol-description": "error",
        "template-curly-spacing": "error",
        "template-tag-spacing": "error",
        "unicode-bom": [
            "error",
            "never"
        ],
        "valid-jsdoc": "off",
        "vars-on-top": "off",
        "wrap-regex": "error",
        "yield-star-spacing": "error",
        "yoda": [
            "error",
            "never"
        ]
    }
};
```
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Travis run checking our js linting. Fails because it thinks that `try { … } catch { … }` is not legal without a variable for the catch. So `try { … } catch (e) { … }` would be ok. I cannot repro on local which is strange but the javascript validator at https://esprima.org/demo/validate.html also agrees that the syntax is not valid.

```js
n/a
```

```bash
./node_modules/.bin/eslint decksite/webpack.config.js shared_web/static/js/charts.js shared_web/static/js/pd.js shared_web/static/js/tooltips.js shared_web/static/js/index.jsx
```

**What did you expect to happen?**

eslint runs to completion

**What actually happened? Please include the actual, raw output from ESLint.**

Syntax error.

```
/home/travis/build/PennyDreadfulMTG/Penny-Dreadful-Tools/node_modules/eslint/bin/eslint.js:93
        } catch {
                ^
SyntaxError: Unexpected token {
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:617:28)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)
    at startup (bootstrap_node.js:204:16)
    at bootstrap_node.js:625:3
CalledProcessError running ['dev.py', 'jslint']:  [(1, ['./node_modules/.bin/eslint', 'decksite/webpack.config.js', 'shared_web/static/js/charts.js', 'shared_web/static/js/pd.js', 'shared_web/static/js/tooltips.js', 'shared_web/static/js/index.jsx'])] Command '['./node_modules/.bin/eslint', 'decksite/webpack.config.js', 'shared_web/static/js/charts.js', 'shared_web/static/js/pd.js', 'shared_web/static/js/tooltips.js', 'shared_web/static/js/index.jsx']' returned non-zero exit status 1.
Traceback (most recent call last):
  File "dev.py", line 356, in <module>
    run()
  File "dev.py", line 31, in run
    run_dangerously()
  File "dev.py", line 71, in run_dangerously
    jslint()
  File "dev.py", line 276, in jslint
    subprocess.check_call(cmd + files)
  File "/opt/python/3.8.0/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['./node_modules/.bin/eslint', 'decksite/webpack.config.js', 'shared_web/static/js/charts.js', 'shared_web/static/js/pd.js', 'shared_web/static/js/tooltips.js', 'shared_web/static/js/index.jsx']' returned non-zero exit status 1.
The command "$CMD" exited with 1.
```

See https://travis-ci.com/github/PennyDreadfulMTG/Penny-Dreadful-Tools/jobs/339454787

**Are you willing to submit a pull request to fix this bug?**

Yes. Following up with this.

**Comment by aladdin-add** [comment]
we don't officially support Node v11, so please consider using:https://github.com/eslint/eslint/blob/d55490fa73ff69416de375e4c1cd67b6edba531c/package.json#L144

**Comment by aladdin-add** [comment]
it surely is a valid JS, please see https://node.green/#ES2019-misc-optional-catch-binding
![image](https://user-images.githubusercontent.com/13050025/82803758-aacb3480-9eb3-11ea-96aa-8179b9275384.png)

**Comment by bakert** [comment]
Ah, I see. We will upgrade our node version on travis. Thank you.

**Comment by sleepq123** [comment]
> eslint v7 does not officially support Node v11, so please consider using:
> 
> https://github.com/eslint/eslint/blob/d55490fa73ff69416de375e4c1cd67b6edba531c/package.json#L144

❤ < 3
